### PR TITLE
[5]-  move system stats plugin to task plugin

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes
 move system stats plugin to task plugin

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: installation/sql/mysql/base.sql [hunk 1]
```diff
@@ -374,6 +374,7 @@ INSERT INTO `#__extensions` (`package_id`, `name`, `type`, `element`, `folder`,
 (0, 'plg_task_sessiongc', 'plugin', 'sessiongc', 'task', 0, 1, 1, 0, 1, '', '{}', '', 7, 0),
 (0, 'plg_task_sitestatus', 'plugin', 'sitestatus', 'task', 0, 1, 1, 0, 1, '', '{}', '', 8, 0),
 (0, 'plg_task_updatenotification', 'plugin', 'updatenotification', 'task', 0, 1, 1, 0, 1, '', '{}', '', 9, 0),
+(0, 'plg_task_stats', 'plugin', 'stats', 'task', 0, 1, 1, 0, 1, '', '{}', '', 10, 0),
```
Comment (richard67) [inline installation/sql/mysql/base.sql]:
```suggestion
(0, 'plg_task_sitestatus', 'plugin', 'sitestatus', 'task', 0, 1, 1, 0, 1, '', '{}', '', 8, 0),
(0, 'plg_task_stats', 'plugin', 'stats', 'task', 0, 1, 1, 0, 1, '', '{}', '', 9, 0),
(0, 'plg_task_updatenotification', 'plugin', 'updatenotification', 'task', 0, 1, 1, 0, 1, '', '{}', '', 10, 0),
```
Please alpha order the plugins of the same folder.

File: installation/sql/postgresql/base.sql [hunk 2]
```diff
@@ -380,6 +380,7 @@ INSERT INTO "#__extensions" ("package_id", "name", "type", "element", "folder",
 (0, 'plg_task_sessiongc', 'plugin', 'sessiongc', 'task', 0, 1, 1, 0, 1, '', '{}', '', 7, 0),
 (0, 'plg_task_sitestatus', 'plugin', 'sitestatus', 'task', 0, 1, 1, 0, 1, '', '{}', '', 8, 0),
 (0, 'plg_task_updatenotification', 'plugin', 'updatenotification', 'task', 0, 1, 1, 0, 1, '', '{}', '', 9, 0),
+(0, 'plg_task_stats', 'plugin', 'stats', 'task', 0, 1, 1, 0, 1, '', '{}', '', 10, 0),
```
Comment (richard67) [inline installation/sql/postgresql/base.sql:383]:
```suggestion
(0, 'plg_task_sitestatus', 'plugin', 'sitestatus', 'task', 0, 1, 1, 0, 1, '', '{}', '', 8, 0),
(0, 'plg_task_stats', 'plugin', 'stats', 'task', 0, 1, 1, 0, 1, '', '{}', '', 9, 0),
(0, 'plg_task_updatenotification', 'plugin', 'updatenotification', 'task', 0, 1, 1, 0, 1, '', '{}', '', 10, 0),
```
Please alpha order the plugins of the same folder.

File: libraries/src/Extension/ExtensionHelper.php [hunk 3]
```diff
@@ -323,6 +322,7 @@ class ExtensionHelper
         ['plugin', 'sessiongc', 'task', 0],
         ['plugin', 'sitestatus', 'task', 0],
         ['plugin', 'updatenotification', 'task', 0],
+        ['plugin', 'stats', 'task', 0],
```
Comment (richard67) [inline libraries/src/Extension/ExtensionHelper.php]:
```suggestion
        ['plugin', 'sitestatus', 'task', 0],
        ['plugin', 'stats', 'task', 0],
        ['plugin', 'updatenotification', 'task', 0],
```
Please alpha order plugins of the same folder.

**Comment by richard67** [comment]
System tests are failing for new installation (regardless of database type). Reason could be that the statistics message is not shown as it should be after a new installation.

**Comment by wilsonge** [comment]
I'd leave this one as a system plugin - doesn't stop you moving the sending of the data to the task events (don't forget plugin groups are just how the plugins are loaded) doesn't stop you using other events.

So leave it as a system - because we still need to get consent from the user - but then use the task infrastructure for sending the data.

**Comment by alikon** [comment]
> we still need to get consent from the user

and we are getting it, cause the user need to schedule that task... so you are more than informed

**Comment by growunder** [comment]
I have tested this item :red_circle: unsuccessfully on 7ed72c044bbfb8214c0e64048a41a914a3e973c5<br /><br />Could not reproduce, Systems Stats remains as stats plugin, not moved to task plugin after patch applied<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43199">issues.joomla.org/tracker/joomla-cms/43199</a>.</sub>

**Comment by richard67** [comment]
@growunder It is not sufficient just to apply the patch, it also needs to run the SQL statement in the update SQL script added by this PR. Or use the packages created by drone for this PR.

@alikon Would be good to have testing instructions so testers know what to do.

**Comment by alikon** [comment]
1st we should get some consensus, and as for now we don't get it.... so

**Pull request closed without merging**